tests/unit/db/result_tests.py

```diff
@@ -88,43 +106,522 @@ def test_constructor(self):
+
+    def test_get_item_by_missing_key(self):
+        """
+        Test retrieving a result using value that refers to a key that does not
+        exist in the result.
+        """
+        result = Result(self.view001)
+        self.assertEqual(result['ruby010'], [])
+        self.assertEqual(result[ResultByKey('ruby010')], [])
+
+    def test_get_item_by_complex_key(self):
+        """
+        Test retrieving a result using value that refers to a complex key of the
+        result.
+        """
+        result = Result(self.view005)
+        expected = [{'key': ['julia', 10], 'id': 'julia010', 'value': 1}]
+        self.assertEqual(result[['julia', 10]], expected)
+        self.assertEqual(result[ResultByKey(['julia', 10])], expected)
+
+    def test_get_item_by_integer_key(self):
+        """
+        Test retrieving a result using an integer value that refers to a key of
+        the result.
+        """
+        result = Result(self.view003)
+        expected = [{'key': 10, 'id': 'julia020', 'value': 1},
+                    {'key': 10, 'id': 'julia021', 'value': 1}]
+        self.assertEqual(result[ResultByKey(10)], expected)
+
+    def test_get_item_by_missing_integer_key(self):
+        """
+        Test retrieving a result using an integer value that refers to a key
+        that does not exist in the result.
+        """
+        result = Result(self.view003)
+        self.assertEqual(result[ResultByKey(99)], [])
+
+    def test_get_item_slice_no_start_no_stop(self):
+        """
+        Test that by not providing a start and a stop slice value, the entire
+        result is returned.
+        """
+        result = Result(self.view001, limit=3)
+        expected = [{'key': 'julia000', 'id': 'julia000', 'value': 1},
+                    {'key': 'julia001', 'id': 'julia001', 'value': 1},
+                    {'key': 'julia002', 'id': 'julia002', 'value': 1}]
+        self.assertEqual(result[:], expected)
+
+    def test_get_item_invalid_index_slice(self):
+        """
+        Test that when invalid start and stop values are provided in a slice
+        an exception is raised.
+        """
+        result = Result(self.view001)
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result[-1: 10]
+        self.assertEqual(cm.exception.status_code, 101)
+
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result[1: -10]
+        self.assertEqual(cm.exception.status_code, 101)
+
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result[-1: -10]
+        self.assertEqual(cm.exception.status_code, 101)
+
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result[5: 2]
+        self.assertEqual(cm.exception.status_code, 101)
+
+    def test_get_item_index_slice_using_start_stop(self):
+        """
+        Test getting an index slice by using start and stop slice values.
+        """
```

Comment:
Good catch.  I updated the code to raise an invalid argument exception in this case and added a test.  Done in 033b46ea1539fa63309b68ca11b7da411ee15744 and e6d1b8b.